packages/untp-test-suite/src/interfaces/utils/common.ts

```diff
@@ -9,4 +18,50 @@ export async function readJsonFile<T>(filePath: string): Promise<T | null> {
   } catch (error) {
     return null;
   }
-}
\ No newline at end of file
+}
+
+export const getTemplateName = (testSuiteResult: IValidatedCredentials): string => {
+  if (testSuiteResult.errors === null) {
+    return TemplateName.pass;
+  }
+
+  if (Array.isArray(testSuiteResult.errors) && testSuiteResult.errors.length > 0) {
+    const getErrors = testSuiteResult.errors as ErrorObject[];
+    const errorWithAdditionalProperty = getErrors.find((error) => error?.params?.additionalProperty);
+    if (errorWithAdditionalProperty) {
+      // If any ErrorObject has 'additionalProperty', return 'warning'
+      return TemplateName.warn;
+    }
+
+    // If no ErrorObject has 'additionalProperty', return 'failed'
+    return TemplateName.failed;
+  }
+
+  return TemplateName.failed;
+};
+
+export const generateFinalMessage = (credentials: ICredentialTestResult[]): IFinalReport => {
+  let finalMessage = TestSuiteMessage.Pass;
+
+  if (credentials.some((credential) => credential.result === TemplateName.failed)) {
+    finalMessage = TestSuiteMessage.Failed;
+  }
+
+  if (credentials.some((credential) => credential.result === TemplateName.warn)) {
+    finalMessage = TestSuiteMessage.Warning;
+  }
+
+  let finalStatus: FinalStatus = 'PASS';
+  if (finalMessage === TestSuiteMessage.Failed) {
+    finalStatus = 'FAILED';
+  }
```

Comment:
Message template:
```
Testing Credential: productPassport v1.0
Path: path/to/credential.json
Result: PASS
---
Testing Credential: objectEvent v2.1
Path: path/to/objectEvent.json
Result: WARN - [Reason for warning (extended the core model)]
---
Testing Credential: aggregationEvent v2.1
Path: path/to/aggregationEvent.json
Result: FAIL - [Error details or reason for failure]
---
```

Final report:
```
|               UNTP Core Test Suite               |
|:------------------------------------------------:|
|               Runner version V0.01               |
| Credential Type        | Version | Status        |
|------------------------|---------|---------------|
| productPassport        | v1.2    | PASS          |
| conformityCredential   | v1.0    | PASS          |
| objectEvent            | v2.1    | PASS          |
| aggregationEvent       | v2.1    | PASS          |
| transformationEvent    | v2.1    | PASS          |
| transactionEvent       | v2.1    | PASS          |
|------------------------|---------|---------------|
|    **Your credentials are UNTP compliant 🎉**    |
```

I think we should follow the template we discussed before.
As you can see, both the message templates and the final report have three result/status types: `PASS`, `WARN`, `FAIL`. So, you can change `FAILED` to `FAIL` to match the template above.